File: mainframe_platform_frontend/src/types/project.ts
import { ApiResponse, Paginate } from "./common";

export interface Project {
  id: string;
  name: string;
  description: string;
  user_id: string;
}

export interface PaginatedProjectsData<T> extends Paginate {
  projects: T[];
}

export type ProjectApiResponse = ApiResponse<PaginatedProjectsData<Project>>;

File: mainframe-services/app/models/project.py
from pydantic import BaseModel, Field
from typing import Optional, List
from bson import ObjectId
from app.schemas.common_schema import PyObjectId


class Project(BaseModel):
    id: PyObjectId = Field(default_factory=PyObjectId, alias="_id")
    name: str
    description: Optional[str] = None
    user_id: PyObjectId

    class Config:
        populate_by_name = True
        arbitrary_types_allowed = True
        json_encoders = {ObjectId: str}
